### PR TITLE
CI build time reduction

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6110)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by ianvernon** [comment]
Wow, huge time reduction ! Good find.

**Comment by tgraf** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -117,6 +117,31 @@ func (kub *Kubectl) CepGet(namespace string, pod string) *models.Endpoint {
 	return data
 }
 
+// WaitForEnforcingCNP waits for the CNP with the specified to go into
+// enforcing mode on all nodes
+func (kub *Kubectl) WaitForEnforcingCNP(cnpName string) error {
+	body := func() bool {
+		for _, node := range []string{K8s1, K8s2} {
```
Comment (eloycoto) [inline test/helpers/kubectl.go]: This will not work on nightly test. We run with 4 nodes.

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -117,6 +117,31 @@ func (kub *Kubectl) CepGet(namespace string, pod string) *models.Endpoint {
...
+			cnpCMD := fmt.Sprintf("%s get cnp %s -o jsonpath='{range .status.nodes.%s}{.enforcing}{end}'", KubectlCmd, cnpName, node)
```
Comment (eloycoto) [inline test/helpers/kubectl.go]: This should get a namespace, multiple test are using a second namespace.

File: test/helpers/kubectl.go [hunk 3]
```diff
@@ -117,6 +117,31 @@ func (kub *Kubectl) CepGet(namespace string, pod string) *models.Endpoint {
...
+			res := kub.Exec(cnpCMD)
+			if !res.WasSuccessful() {
+				return false
+			}
+
+			if strings.TrimSuffix(res.Output().String(), "\n") != "true" {
```
Comment (eloycoto) [inline test/helpers/kubectl.go]: `res.SingleOut() != true`

File: test/helpers/kubectl.go (hunk 2 above)
Comment (eloycoto) [reply test/helpers/kubectl.go]: Why not range over nodes and get only one time and validate that all are enforcing?

**Comment by tgraf** [comment]
@eloycoto Great feedback, thanks! I'm mostly still testing. Also, need to find a better CEPReady alternative for the KNP tests as it adds ~30s for each test. I will clean this up and propose it for merging as it already reduces the time significantly.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-missed-k8s

**Comment by tgraf** [comment]
test-nightly

**Review by aanm — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Comment by tgraf** [comment]
test-missed-k8s

**Review by eloycoto — CHANGES_REQUESTED** [review]
Hi,

I'm a bit against of this PR:

- Microscope is disabled, I'm happy with that, but that will be enabled in
  future, are we tracking this down?

- These changes are now no testing the CEP resources, and that is a user
  feature, for me, it shouldn't be removed, is a user interface that need to be
  tested. IMHO

- The WaitForEnforcingCNP function is nice, but.. that is what
  CiliumPolicyAction is doing. So I'm not sure what is the value added.


Finally, the reason to have CEP was not to wait to enforce, the reason is to
validated that the CEP was working correctly.

Regards

**Comment by tgraf** [comment]
> * Microscope is disabled, I'm happy with that, but that will be enabled in
>   future, are we tracking this down?

Yes, I will create an issue once this is merged. In its current form, it is absolutely unacceptable to run. I left all of the code in there so we can just re-enable via changing the variable.

> * These changes are now no testing the CEP resources, and that is a user
>   feature, for me, it shouldn't be removed, is a user interface that need to be
>   tested. IMHO

The existing logic was flawed. The details are in the commit message. There is a lot of logic left to validate the CEP update. I don't think we need to validate that logic over and over again in every test. In fact, the logic of the enforcing flag, which is the designated function to determine whether a policy is being enforced was not being validated at all.

> * The WaitForEnforcingCNP function is nice, but.. that is what
>   CiliumPolicyAction is doing. So I'm not sure what is the value added.

It is not the same logic. We'll have to revisit that logic as it seems to do the following:
* Retrieve the policy revision of all agents
* Import the policy
* Wait for all agents to advance past the previously retrieved policy revision

It does almost the same but it does not guarantee the new CNP at all as the policy revision can be bumped on an agent between retrieving the revision and importing the policy and we would declare enforcement to early. There are many other places using CIliumAction that are not using the new WaitForEnforcingCNP yet so I left it in place for now.

**Comment by tgraf** [comment]
@eloycoto The point of testing CEP propagation is fair but should test that in a dedicated test and not pay the cost for the propagation delay in every single policy related test. I'll take care of adding a GH issue to track the addition. For now, the PR only removes a small fraction of the CEP checking so there is still good coverage overall.

**Comment by tgraf** [comment]
The following tests failed:
- Nightly: One test failed due to etcd operator scaling down, all policy tests succeeded
- test-missed-k8s: One test failed because of `Failed to replace Qdisc for lxcbcee7644f6b7`

All tests affected by this PR passed.

**Comment by ianvernon** [comment]
I'd like to backport the microscope disabling to the v1.3 branch; nominating the microscope-related commits for backport.

**Comment by ianvernon** [comment]
Note regarding backport of this to v1.3: b9a199919a51d7671b2258c08846c52c0756b89b and  2be54f5039965be47da617ff5b6c98bccefffd79 are being backported to disable microscope in the v1.3 CI. The other commit ( eef2d873c9d7f17135a54e29076b2b5b3e7a5879 ) is not being backported since it is not related to microscope disablement.

**Pull request merged**